ODIN_II/SRC/include/MixingOptimization.hpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#ifndef MIXING_OPTIMIZATION_HPP
+#define MIXING_OPTIMIZATION_HPP
+#include "odin_types.h" // netlist_t, config_t
+/**
+ * @brief A base class in hierarchy for complex synthesis
+ * allowing for mixing soft and hard logic 
+ */
+class MixingOpt {
+  public:
+    /**
+     * @brief Construct a new Mixing Opt object
+     * 
+     * By default, all optimizations only share 
+     * the ratio of blocks to be implemented in 
+     * hard logic
+     * @param ratio, a value within 0 to 1 to 
+     * implement ratio*requested hard blocks in
+     * hard logic
+     * @param kind a kind of blocks that correspond
+     * to optimization pass
+     */
+    MixingOpt(float ratio, operation_list kind)
+        : _kind(kind) {
+        _ratio = ratio;
+        _enabled = true;
```

Comment:
set to false by default

Reply:
it is set to false by default, and only if this constructor is called, the _enabled is set to true.

Reply:
I would rather this class don't touch it then and leave it false, only the implemented classes should change this, it is more explicit.

Reply:
Done, thanks for the comment!

Reply:
Done, thanks for the comment!